cmd/create/cluster/cmd.go

```diff
@@ -418,40 +399,98 @@ func run(cmd *cobra.Command, _ []string) {
 
 	// AWS ARN Role
 	roleARN := args.roleARN
-	if roleARN != "" && interactive.Enabled() {
+
+	if !interactive.Enabled() && creator.IsSTS && roleARN == "" {
+		err = interactive.PrintHelp(interactive.Help{
+			Message: "Since your AWS credentials are returning an STS ARN you can only " +
+				"create STS clusters. Otherwise, switch to IAM credentials.",
+		})
+		if err != nil {
+			reporter.Errorf("%s", err)
+			os.Exit(1)
+		}
+		interactive.Enable()
+		reporter.Infof("Enabling interactive mode")
+	}
+
+	if interactive.Enabled() {
 		roleARN, err = interactive.GetString(interactive.Input{
 			Question: "Role ARN",
 			Help:     cmd.Flags().Lookup("role-arn").Usage,
 			Default:  roleARN,
+			Required: creator.IsSTS,
 		})
 		if err != nil {
 			reporter.Errorf("Expected a valid ARN: %s", err)
 			os.Exit(1)
 		}
 	}
+	if roleARN != "" {
+		_, err = arn.Parse(roleARN)
+		if err != nil {
+			reporter.Errorf("Expected a valid Role ARN: %s", err)
+			os.Exit(1)
+		}
+	} else {
+		aws.CheckStackReadyForCreateCluster(reporter, logger)
+	}
 
-	// All STS-related options
 	externalID := args.externalID
-	tagsList := map[string]string{}
-	operatorIAMRoleList := []clusterprovider.OperatorIAMRole{}
-	masterRoleARN := ""
-	workerRoleARN := ""
+	if roleARN != "" && interactive.Enabled() {
+		externalID, err = interactive.GetString(interactive.Input{
+			Question: "External ID",
+			Help:     cmd.Flags().Lookup("external-id").Usage,
+		})
+		if err != nil {
+			reporter.Errorf("Expected a valid External ID: %s", err)
+			os.Exit(1)
+		}
+	}
 
-	if roleARN != "" {
-		if interactive.Enabled() {
-			externalID, err = interactive.GetString(interactive.Input{
-				Question: "External ID",
-				Help:     cmd.Flags().Lookup("external-id").Usage,
-			})
-			if err != nil {
-				reporter.Errorf("Expected a valid External ID: %s", err)
-				os.Exit(1)
-			}
+	// OpenShift version:
+	version := args.version
+	channelGroup := args.channelGroup
+	versionList, err := getVersionList(ocmClient, channelGroup)
+	if err != nil {
+		reporter.Errorf("%s", err)
+		os.Exit(1)
+	}
+	if version == "" {
+		version = versionList[0]
+	}
+	if interactive.Enabled() {
+		version, err = interactive.GetOption(interactive.Input{
+			Question: "OpenShift version",
+			Help:     cmd.Flags().Lookup("version").Usage,
+			Options:  versionList,
+			Default:  version,
+		})
+		if err != nil {
+			reporter.Errorf("Expected a valid OpenShift version: %s", err)
+			os.Exit(1)
 		}
+	}
+	version, err = validateVersion(version, versionList)
+	if err != nil {
+		reporter.Errorf("Expected a valid OpenShift version: %s", err)
+		os.Exit(1)
+	}
 
-		operatorIAMRoles := args.operatorIAMRoles
+	operatorIAMRoles := args.operatorIAMRoles
+	operatorIAMRoleList := []clusterprovider.OperatorIAMRole{}
+	if roleARN != "" {
 		for _, role := range operatorIAMRoles {
+			if !strings.Contains(role, ",") {
+				reporter.Errorf("Expected operator IAM roles to be a comma-separated " +
```

Comment:
Use the flag usage here?

Reply:
Maybe, but this explains the syntax, whereas the flag usage explains the purpose. The flag usage already shows up in interactive mode when pressing `?`